src/components/interface/ResolutionWarning.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  Modal, Text, Title, Stack,
+} from '@mantine/core';
+import { useEffect, useState, useRef } from 'react';
+import { IconAlertTriangle } from '@tabler/icons-react';
+import { useNavigate } from 'react-router';
+import { useStoreSelector } from '../../store/store';
+import { useStorageEngine } from '../../storage/storageEngineHooks';
+import { useStudyId } from '../../routes/utils';
+
+export function ResolutionWarning() {
+  const studyConfig = useStoreSelector((state) => state.config);
```

Comment:
Better to use our hook
```suggestion
  const studyConfig = useStudyConfig();
```